mrfCommon/src/flash.cpp

```diff
@@ -371,14 +371,16 @@ void CFIFlash::busyWait(double timeout, unsigned n)
 CFIStreamBuf::CFIStreamBuf(CFIFlash& flash)
     :flash(flash)
     ,pos(0u)
-{}
+{
+    buf.resize(1);
+}
 
 CFIStreamBuf::int_type CFIStreamBuf::underflow()
 {
     // read-ahead is only one page
     buf.resize(flash.pageSize());
     flash.read(pos, buf.size(), (epicsUInt8*)&buf[0]);
-    setg(&buf[0], &buf[0], &buf[buf.size()]);
+    setg(&buf[0], &buf[0], &buf[buf.size()-1]);
```

Comment:
As prudent paranoia, I would suggest ensuring that `pageSize()` can never be zero.

eg. maybe in `CFIFlash::check()()`.

or changing the resize immediately above to:

```c++
buf.resize(std::min(1u, flash.pageSize()));
```

Reply:
Good idea! 
I think you mean std::max right ?

Reply:
Change applied.

Reply:
> I think you mean std::max right ?

My thinking is to ensure that `size()>=1` so that `buf.size()-1u` will never underflow..  So `min()`.

Reply:
Sorry, but I might be missing something, if we do:
`buf.resize(std::min(1u, flash.pageSize()));`

And the size is 0, the min will choose 0. And if the size > 1 it will resize to 1.

Using max, if size is 0, it will choose 1, so the `buf.size()-1u` will not underflow.
And if size > 1, it will choose 0.

Reply:
Ah, ok.  You are right and I am wrong.  max() it is.